src/Storages/MergeTree/IMergeTreeDataPartWriter.cpp

```diff
@@ -116,19 +122,21 @@ SerializationPtr IMergeTreeDataPartWriter::getSerialization(const String & colum
 
 ASTPtr IMergeTreeDataPartWriter::getCodecDescOrDefault(const String & column_name, CompressionCodecPtr default_codec) const
 {
-    auto get_codec_or_default = [&](const auto & column_desc)
-    {
-        return column_desc.codec ? column_desc.codec : default_codec->getFullCodecDesc();
-    };
+    ASTPtr default_codec_desc = default_codec->getFullCodecDesc();
+
+    auto default_compression_codec_mergetree_settings = (*storage_settings)[MergeTreeSetting::default_compression_codec].toString();
```

Comment:
Shouldn't this work? Then let's use to properly match the type.

```suggestion
    auto default_compression_codec_mergetree_settings = (*storage_settings)[MergeTreeSetting::default_compression_codec].value;
```